libvcx/src/api_lib/api_handle/wallet.rs

```diff
@@ -0,0 +1,115 @@
+use aries_vcx::common::signing::unpack_message_to_string;
+use aries_vcx::errors::error::VcxResult;
+use aries_vcx::indy;
+use aries_vcx::indy::wallet::{close_search_wallet, fetch_next_records_wallet, import, IssuerConfig, open_search_wallet, RestoreWalletConfigs};
+use aries_vcx::protocols::connection::pairwise_info::PairwiseInfo;
+use aries_vcx::vdrtools::SearchHandle;
+
+use crate::api_lib::errors::error::LibvcxResult;
+use crate::api_lib::errors::mapping_from_ariesvcx::map_ariesvcx_result;
+use crate::api_lib::global::profile::{get_main_profile, get_main_wallet};
+use crate::api_lib::global::wallet::get_main_wallet_handle;
+
+pub async fn key_for_local_did(did: &str) -> LibvcxResult<String> {
+    let wallet = get_main_wallet();
+    map_ariesvcx_result(wallet.key_for_local_did(did).await)
+}
+
+pub async fn wallet_sign(vk: &str, data_raw: &[u8]) -> LibvcxResult<Vec<u8>> {
+    let wallet = get_main_wallet();
+    map_ariesvcx_result(wallet.sign(&vk, &data_raw).await)
+}
+
+pub async fn wallet_verify(vk: &str, msg: &[u8], signature: &[u8]) -> LibvcxResult<bool> {
+    let wallet = get_main_wallet();
+    map_ariesvcx_result(wallet.verify(vk, msg, signature).await)
+}
+
+pub async fn replace_did_keys_start(did: &str) -> LibvcxResult<String> {
+    let wallet = get_main_wallet();
+    map_ariesvcx_result(wallet.replace_did_keys_start(&did).await)
+}
+
+pub async fn rotate_verkey_apply(did: &str, temp_vk: &str) -> LibvcxResult<()> {
+    let profile = match get_main_profile() {
```

Comment:
sorry, very nitpicky, but can this just be `get_main_profile()?`

Reply:
Happy about nitpicking no worries :-D Fixed this anti-pattern across the board.